src/lib/get-costume-url.js

```diff
@@ -0,0 +1,31 @@
+import {SVGRenderer} from 'scratch-svg-renderer';
+
+// Contains 'font-family', but doesn't only contain 'font-family="none"'
+const HAS_FONT_REGEXP = 'font-family(?!="none")';
+
+const getCostumeUrl = function (assetId, vm) {
+
+    const storage = vm.runtime.storage;
+    const asset = storage.get(assetId);
+    // If the SVG refers to fonts, they must be inlined in order to display correctly in the img tag.
+    // Avoid parsing the SVG when possible, since it's expensive.
+    if (asset.assetType === storage.AssetType.ImageVector) {
+        // If the asset ID has not changed, no need to re-parse
+
+        const svgRenderer = new SVGRenderer();
+
+        const svgString = vm.runtime.storage.get(assetId).decodeText();
```

Comment:
Whenever a sprite image updates, this expensive decodeText and loadString operation gets called twice with the same assetId, once on the sprite selector in the target info pane and once on the watermark. Since they share code here it might be possible to share caches too, saving an unnecessary duplicate decode. (Just a thought, not necessarily for this PR)

Reply:
[slow costume update.zip](https://github.com/LLK/scratch-gui/files/2473747/slow.costume.update.zip)

Here's an example project where you can hopefully see the slowdown between this branch and develop (rename to .sb3)